### PR TITLE
IRGen and runtime support for key path resilience.

### PR DESCRIPTION
**Pull request by jckarter**
IRGen support for emitting computed key path components with references to external property descriptors, and matching runtime support for instantiating references to external property descriptors into key path objects.

Includes a fix for rdar://problem/39196876.

### CODE REVIEW

File: test/SILGen/keypaths.swift [hunk 1]
```diff
@@ -1,5 +1,4 @@
-
-// RUN: %target-swift-emit-silgen -module-name keypaths -enable-sil-ownership %s | %FileCheck %s
+// RUN: %target-swift-emit-silgen -module-name keypaths %s | %FileCheck %s
```
Comment (slavapestov) [inline test/SILGen/keypaths.swift]: Is this always on by default now?

Comment (jckarter) [reply test/SILGen/keypaths.swift]: I believe so, yeah. cc @gottesmm

**Comment by jckarter** [comment]
@swift-ci Please test

**Comment by jckarter** [comment]
@swift-ci Please test source compatibility

**Comment by jckarter** [comment]
@swift-ci Please test

**Comment by jckarter** [comment]
@swift-ci Please test

**Pull request merged**